iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,342 @@
+    EXPECT_EQ(true, interfacePortData->m_doInitialOfferForward);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareInterface_InterfacelistOverflow)
+{
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    for (auto i = 0u; i < iox::MAX_INTERFACE_NUMBER; ++i)
+    {
+        auto interfacePort = m_runtime->getMiddlewareInterface(iox::capro::Interfaces::INTERNAL);
+        EXPECT_NE(nullptr, interfacePort);
+    }
+    auto interfacePort = m_runtime->getMiddlewareInterface(iox::capro::Interfaces::INTERNAL);
+
+    EXPECT_EQ(nullptr, interfacePort);
+    EXPECT_TRUE(errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, SendMessageToRouDi_ValidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << m_runtimeName
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_runnableName;
+
+    const auto successfullySent = m_runtime->sendMessageToRouDi(m_sendBuffer);
+
+    EXPECT_EQ(true, successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, SendMessageToRouDi_InvalidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << std::string()
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_invalidRunnableName;
+
+    const auto successfullySent = m_runtime->sendMessageToRouDi(m_sendBuffer);
+
+    EXPECT_EQ(false, successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, SendMessageToRouDi_EmptyMessage)
+{
+    const auto successfullySent = m_runtime->sendMessageToRouDi(m_sendBuffer);
+
+    EXPECT_EQ(true, successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, SendRequestToRouDi_ValidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << m_runtimeName
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_runnableName;
+
+    const auto successfullySent = m_runtime->sendRequestToRouDi(m_sendBuffer, m_receiveBuffer);
+
+    EXPECT_TRUE(m_receiveBuffer.isValid());
+    EXPECT_EQ(true, successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, SendRequestToRouDi_InvalidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << m_runtimeName
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_invalidRunnableName;
+
+    const auto successfullySent = m_runtime->sendRequestToRouDi(m_sendBuffer, m_receiveBuffer);
+
+    EXPECT_EQ(false, successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSenderIsSuccessful)
+{
+    const auto senderPort = m_runtime->getMiddlewareSender(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), senderPort->m_serviceDescription);
```

Comment:
please add an `ASSERT_THAT(senderPort, Ne(nullptr))` before you access the members of the senderPort. This could be a nullptr and then the test would crash.

Please also do it everywhere you call getMiddlewareX and then de-reference the returned pointer

Reply:
done